aws/resource_aws_sagemaker_endpoint_configuration_test.go

```diff
@@ -0,0 +1,551 @@
+					testAccCheckSagemakerEndpointConfigExists("aws_sagemaker_endpoint_configuration.foo"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_endpoint_configuration.foo",
+						"production_variants.0.accelerator_type",
+						"ml.eia1.medium"),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_endpoint_configuration.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerEndpointConfig_kmsKeyId(t *testing.T) {
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerEndpointConfigDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerEndpointConfigKmsKeyIdConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerEndpointConfigExists("aws_sagemaker_endpoint_configuration.foo"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_endpoint_configuration.foo", "kms_key_id"),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_endpoint_configuration.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerEndpointConfig_tags(t *testing.T) {
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerEndpointConfigDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerEndpointConfigConfigTags(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerEndpointConfigExists("aws_sagemaker_endpoint_configuration.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint_configuration.foo",
+						"tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint_configuration.foo",
+						"tags.foo", "bar"),
+				),
+			},
+			{
+				Config: testAccSagemakerEndpointConfigTagsUpdateConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerEndpointConfigExists("aws_sagemaker_endpoint_configuration.foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint_configuration.foo",
+						"tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_endpoint_configuration.foo",
+						"tags.bar", "baz"),
+				),
+			},
+			{
+				ResourceName:      "aws_sagemaker_endpoint_configuration.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckSagemakerEndpointConfigDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).sagemakerconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_sagemaker_endpoint_configuration" {
+			continue
+		}
+
+		resp, err := conn.ListEndpointConfigs(&sagemaker.ListEndpointConfigsInput{
+			NameContains: aws.String(rs.Primary.ID),
```

Comment:
This seems like it can report false positives and should use `DescribeEndpointConfig` instead, e.g.

```go
input := &sagemaker.DescribeEndpointConfigInput{
	EndpointConfigName: aws.String(rs.Primary.ID),
}

output, err := conn.DescribeEndpointConfig(input)

if isAWSErr(err, sagemaker.ErrCodeResourceNotFound, "") {
  continue
}

if err != nil {
  return err
}

return fmt.Errorf("Sagemaker Endpoint Configuration (%s) still exists", rs.Primary.ID)
```